### PR TITLE
update provisioning to run all minor versions by default

### PR DESCRIPTION
**Pull request by slickwarren**
## Issue: 

 https://github.com/rancher/qa-tasks/issues/1145
## Problem

 we want to test all available k8s versions by default, but this is not current standard in the code. 

## Solution

set kubernetes versions == the full returned list from rancher if we don't have any specified in the config.yaml
supporting shepherd PR: https://github.com/rancher/shepherd/pull/256 

## Testing

tested locally and in jenkins

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: go.mod
Comment (floatingman) [inline go.mod:19]: Be sure to change back to rancher/shepherd before merging

File: tests/v2/validation/provisioning/registries/registry_test.go
Comment (floatingman) [inline tests/v2/validation/provisioning/registries/registry_test.go:251]: What's the reason for adding the space at the end?

File: tests/v2/validation/provisioning/registries/registry_test.go
Comment (floatingman) [inline tests/v2/validation/provisioning/registries/registry_test.go:326]: What's the reason for the space at the end of the string?

File: tests/v2/validation/provisioning/registries/registry_test.go
Comment (floatingman) [inline tests/v2/validation/provisioning/registries/registry_test.go:393]: What's the reason for the space at the end of the string?

**Review by floatingman — CHANGES_REQUESTED** [review]
Looks good. Just a few comments.

**Review by anupama2501 — COMMENTED** [review]
Could you rebase the PR as some of the changes in the PR are moved to actions F.E "github.com/rancher/shepherd/extensions/provisioninginput" is moved to actions, it will cause some conflicts. Also, please change it to draft if possible as it has some shepherd changes.

File: tests/v2/validation/provisioning/registries/registry_test.go
Comment (markusewalker) [reply tests/v2/validation/provisioning/registries/registry_test.go:251]: I'm pretty sure this is for Qase and the naming.

File: tests/v2/validation/provisioning/k3s/custom_cluster_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/k3s/custom_cluster_test.go:47]: So, is this not going to allow the default version to be selected if none is picked? Or does the list function just run all of the versions? I have this same question for all of the other files.

File: tests/v2/validation/provisioning/rke2/agent_customization_test.go
Comment (markusewalker) [inline tests/v2/validation/provisioning/rke2/agent_customization_test.go:47]: Why is this one being left alone to have the default version?

**Review by markusewalker — COMMENTED** [review]
Left some questions to be clarified. Additionally, I'd like to request Hostbusters gets our Windows PR that's been waiting for code freeze to be lifted to get merged first before we merge this.

File: tests/v2/validation/provisioning/k3s/custom_cluster_test.go
Comment (slickwarren) [reply tests/v2/validation/provisioning/k3s/custom_cluster_test.go:47]: so if you don't specify a k8s verison in your yaml, that's when we want to use the available versions from rancher. The if statement is there to allow for reading from the config.

File: tests/v2/validation/provisioning/registries/registry_test.go
Comment (slickwarren) [reply tests/v2/validation/provisioning/registries/registry_test.go:393]: we append the k8sVersion later, so putting a space here for readability. I could add it later, but its a little prettier to have it here vs. a ' ' on line 398

File: tests/v2/validation/provisioning/registries/registry_test.go
Comment (slickwarren) [reply tests/v2/validation/provisioning/registries/registry_test.go:85]: if this is needed, can it be in a separate PR? The format currently is the same in pretty much all provisioning setup functions.

File: tests/v2/validation/provisioning/rke2/agent_customization_test.go
Comment (slickwarren) [reply tests/v2/validation/provisioning/rke2/agent_customization_test.go:47]: agentCustomization shouldn't be impacted by k8s version minor bumps. It seemed like overkill to run it on all available k8s versions but if you want it done that way I can ruit on all available rke2 types.

File: tests/v2/validation/provisioning/registries/registry_test.go
Comment (slickwarren) [reply tests/v2/validation/provisioning/registries/registry_test.go:251]: correct

File: tests/v2/validation/provisioning/rke2/agent_customization_test.go
Comment (markusewalker) [reply tests/v2/validation/provisioning/rke2/agent_customization_test.go]: No need, thanks for the clarification.

File: tests/v2/validation/provisioning/k3s/custom_cluster_test.go
Comment (markusewalker) [reply tests/v2/validation/provisioning/k3s/custom_cluster_test.go:47]: I see, can we test both of these scenarios just to make sure all is good?

**Review by floatingman — APPROVED** [review]
LGTM

**Pull request merged**
